nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -111,32 +109,6 @@ impl DataStore {
             .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
     }
 
-    /// Looks up an IP pool by name if it does not conflict with your current scope.
-    pub(crate) async fn ip_pools_fetch(
-        &self,
-        opctx: &OpContext,
-        name: &Name,
-    ) -> LookupResult<IpPool> {
-        let authz_silo_id = opctx.authn.silo_required()?.id();
-
-        let (.., authz_pool, pool) = LookupPath::new(opctx, &self)
-            .ip_pool_name(&name)
-            // any authenticated user can CreateChild on an IP pool. this is
-            // meant to represent allocating an IP
-            .fetch_for(authz::Action::CreateChild)
-            .await?;
-
-        // You can't look up a pool by name if it conflicts with your current
-        // scope, i.e., if it has a silo it is different from your current silo
-        if let Some(pool_silo_id) = pool.silo_id {
-            if pool_silo_id != authz_silo_id {
-                return Err(authz_pool.not_found());
-            }
-        }
-
-        Ok(pool)
-    }
-
```

Comment:
This did not make sense as a function here. All the logic was related to the calling context.